misk-aws/src/main/kotlin/misk/jobqueue/sqs/QueueResolver.kt

```diff
@@ -17,25 +17,35 @@ internal class QueueResolver @Inject internal constructor(
   private val currentAccount: AwsAccountId,
   private val defaultSQS: AmazonSQS,
   private val crossRegionSQS: Map<AwsRegion, AmazonSQS>,
+  @ForSqsReceiving private val defaultForReceivingSQS: AmazonSQS,
+  @ForSqsReceiving private val crossRegionForReceivingSQS: Map<AwsRegion, AmazonSQS>,
   private val externalQueues: Map<QueueName, AwsSqsQueueConfig>
 ) {
   private val mapping = ConcurrentHashMap<QueueName, ResolvedQueue>()
 
   operator fun get(q: QueueName): ResolvedQueue {
-    return mapping.computeIfAbsent(q) { resolve(it) }
+    return mapping.computeIfAbsent(q) { resolve(it, false) }
```

Comment:
Do we want separate `mapping` instances for the two variations (for receiving or for sending)

Reply:
(I'm asking because the "for sending" input is only used in the resolution within `computeIfAbsent`, and not as an additional input to the map key during lookups by queue name only)

Reply:
oh yeah this is totally busted, good point

Reply:
cool, glad I didn't totally miss something obvious

Reply:
updated, how about that?